sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -325,6 +325,7 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
         localSparkSession.conf.set(SQLConf.CROSS_JOINS_ENABLED.key, true)
         localSparkSession.conf.set(SQLConf.ANSI_ENABLED.key, true)
         localSparkSession.conf.set(SQLConf.DIALECT.key, SQLConf.Dialect.POSTGRESQL.toString)
+        localSparkSession.conf.set(SQLConf.LEGACY_FROM_DAYTIME_STRING.key, true)
```

Comment:
I think we should go to the legacy path if pgsql dialect is on.

Reply:
Do you propose to use the dialect config instead of this one?

Reply:
We need both:
```
if (pgsql dialect) {
  go legacy path
} else if (legacy config set) {
 go legacy path
} else {
  go new path
}
```